chatminer/visualizations.py

```diff
@@ -124,23 +192,23 @@ def wordcloud(df, ax=None, stopwords=None, authors=None, **kwargs):
 
 
 def calendar_heatmap(
-    df,
-    year,
-    vmin=None,
-    vmax=None,
-    cmap="Blues",
-    fillcolor="whitesmoke",
-    linewidth=1,
-    linecolor=None,
-    daylabels=calendar.day_abbr[:],
-    dayticks=True,
-    monthlabels=calendar.month_abbr[1:],
-    monthticks=True,
-    monthly_border=False,
-    ax=None,
-    authors=None,
-    **kwargs,
-):
+    df: pd.DataFrame,
+    year: int,
+    vmin: IntSeries | None = None,
+    vmax: IntSeries | None = None,
```

Comment:
don't we need simple integers here?

Reply:
I assumed from
```py
if vmin is None:
        vmin = df.min()
    if vmax is None:
        vmax = df.max()
```
That `vmin` and `vmax` are of type Series[int] as that is what `df.min()` and `df.max` return (see following method signature):
```py
(method) def min(
    axis: Axis | None = ...,
    skipna: _bool | None = ...,
    level: None = ...,
    numeric_only: _bool = ...,
    **kwargs: Unknown
) -> Series[Unknown]
```
However I didn't take into account that later in the function these these values are passed into `pc = ax.pcolormesh()` as floats:
```py
pc = ax.pcolormesh(plot_data, vmin=vmin, vmax=vmax, cmap=cmap, **kwargs)
```
Strangely, they are later treated as Series (or at least some type of iterable):
```py
plt.colorbar(
        pc, cax=cax, ticks=[min(vmin), int((min(vmin) + max(vmax)) / 2), max(vmax)]
    )
```

Reply:
You're right, this should be rectified. I think overhauling this needs a dedicated issue and PR. We can ignore it here